lib/i18n-transform.js

```diff
@@ -52,93 +52,172 @@ var _getLanguageFromI18n = function(i18nObj) {
+            return fields ? _extractPreferredLanguagePerField(source, i18nArray, languages, fields)
+                : _extractPreferredLanguage(source, i18nArray, languages);
         }
-    };
+    },
 
-module.exports.transform = function(source, languages){
-    var i18nInfo = _getI18n(source);
+    _transform = function (source, languages, fields) {
+        var i18nInfo = _getI18n(source);
 
-    if(!i18nInfo){
-        return source;
-    }
+        if (!i18nInfo) {
+            return source;
+        }
 
-    var selectedLanguage = _getSelectedLanguage(source, i18nInfo.value, languages);
-    if (!selectedLanguage){
-        return;
-    }
+        var selectedLanguage = _getSelectedLanguage(source, i18nInfo.value, languages, fields);
+        if (!selectedLanguage) {
+            return;
+        }
 
-    var result = _mergeSelectedLanguage(source, selectedLanguage);
-    delete result[i18nInfo.name];
-    return result;
-};
+        var result = _mergeSelectedLanguage(source, selectedLanguage);
+        delete result[i18nInfo.name];
+        return result;
+    },
 
-module.exports.transformDestination = function(source, destinationToAddLanguageTo, languages, callback){
-    var i18nInfo = _getI18n(source);
+    _transformDestination = function (source, destinationToAddLanguageTo, languages, callback, fields) {
+        var i18nInfo = _getI18n(source);
 
-    if(!i18nInfo){
-        return source;
-    }
+        if (!i18nInfo) {
+            return source;
+        }
+
+        var selectedLanguage = _getSelectedLanguage(source, i18nInfo.value, languages, fields);
 
-    var selectedLanguage = _getSelectedLanguage(source, i18nInfo.value, languages);
+        if (!selectedLanguage) {
+            if (callback) {
+                callback(new Error("Could not find desired language"));
+            }
+            return;
+        }
+
+        _mergeSelectedLanguage(destinationToAddLanguageTo, selectedLanguage);
 
-    if (!selectedLanguage) {
         if (callback) {
-            callback(new Error("Could not find desired language"));
+            callback();
         }
-        return;
-    }
+    };
 
-    _mergeSelectedLanguage(destinationToAddLanguageTo, selectedLanguage);
+module.exports.transformByField = function (source, languages, fields) {
+    return _transform(source, languages, fields);
+};
 
-    if (callback) {
-        callback();
-    }
+module.exports.transform = function (source, languages) {
+    return _transform(source, languages);
+};
+
+module.exports.transformByFieldDestination = function (source, destinationToAddLanguageTo, languages, fields, callback) {
+    return _transformDestination(source, destinationToAddLanguageTo, languages, callback, fields);
+};
+
+module.exports.transformDestination = function (source, destinationToAddLanguageTo, languages, callback) {
+    return _transformDestination(source, destinationToAddLanguageTo, languages, callback);
```

Comment:
The signature of _transformDestination has changed. I don't want to break it for existing consumers so I used this mechanism to control the input parameters